frontend/app/helpers/spree/frontend_helper.rb

```diff
@@ -162,6 +162,8 @@ def permitted_product_params
     end
 
     def carousel_image_source_set(image)
+      return '' unless image
```

Comment:
This method is being called only in one place where we check that `image` is present. Here we have to do the same.
https://github.com/spark-solutions/spree/blob/fcfc476c3f59389b94ef001aaf4eeb1d787cf2b3/frontend/app/helpers/spree/frontend_helper.rb#L111-L124